### DISCUSSION TITLE
Can I call swell.filters.list() on products from the backend-api?

### DISCUSSION THREAD

**Discussion by coderreco** [swellstores/community#317] [Help] [answered]
**Nextjs**
**swell-js v3.22.0**

I am trying to list the filters for my products, and I am retrieving my products using the backend API.

When I call swell.products.filters(products) I only get null as a response. If I call swell.products.filters(products.results) I get the following response:
```
Array(10) [
  {
    id: 'price',
    label: 'Price',
    type: 'range',
    options: [ { value: 20, label: 20 }, { value: 330, label: 330 } ],
    interval: 10
  }, { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] },
  { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] }, { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] },
  { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] }, { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] },
  { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] }, { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] },
  { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] }, { id: undefined, label: undefined, type: 'select', options: [ { value: undefined, label: undefined } ] },
  
]
```

Can you not fetch filters off of products from the backend api? Otherwise what am I doing wrong?

Thanks in advance!

**Comment by sags95** [comment]
Hey @coderreco , what does your `products` var contain? Is this just a list of products that are created with `await swell.products.list()` ?

There are also a number of undefined attributes, so might be worth seeing if there was anything misconfigured there.

**Reply by coderreco** [reply]
Hey @sags95, products contains a list of products using swell-node. See here:
```
// node.js
const storeId = process.env.NEXT_PUBLIC_SWELL_STORE_ID
const secretKey = process.env.NEXT_PUBLIC_SWELL_SECRET_KEY

const swellnode = require('swell-node').init(storeId, secretKey)

export { swellnode }  

// products.js
import { swellnode } from './node'

const products = await swellnode.get('/products', {
    where: { active: true },
    limit: 1000,
    expand: ['variants:25'],
  })
```
I have more than 200 products I need to fetch, so using the backend api seems like a better option.

I can check around the attributes to make sure there was no mistakes in configuration, but some added context is I was using the frontend api until I realized that it would only list filters for the products passed in. When I was passing in products from `swell.products.list()` it was working as expected.

**Reply by coderreco** [reply]
Hey @sags95 , any suggestions on this?

**Reply by sags95** [reply, marked as answer]
Hey @coderreco , ah I see. The `swell.products.list()` method is intended to be used with a product list derived from the Frontend API as well. With the Backend API there isn't a way to create a filter list like you can with the Frontend API. I suppose one way would be to fetch the `/attributes` that have the `filterable: true` field to build this list.
